src/main/java/com/onaio/steps/activities/NewHouseholdActivity.java

```diff
@@ -72,20 +85,45 @@ private void populateDataFromIntent() {
     }
 
     public void doneBtnClicked(View view) {
+        if (!progressDialog.isShowing()) {
+            progressDialog.show();
+        }
+
+        saveHousehold();
+    }
+
+    public void cancel(View view){
+        finish();
+    }
+
+    public void saveHousehold() {
         try {
+            ODKBlankForm odkBlankForm = (ODKBlankForm) ODKBlankForm.find(this, getFormId());
             Intent intent = this.getIntent();
             Household household = new HouseholdViewWrapper(this).getHousehold(R.id.generated_household_id, R.id.household_number,R.id.household_comments);
+            household.setOdkJrFormId(odkBlankForm.getJrFormId());
+            household.setOdkJrFormTitle(odkBlankForm.getDisplayName());
```

Comment:
Add the appropriate tests for this section. 


It's critical functionality for Steps and not easily testable or verifiable